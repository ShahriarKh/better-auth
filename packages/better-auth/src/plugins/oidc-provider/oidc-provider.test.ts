import { describe, expect, it } from "vitest";
import { betterAuth } from "../../auth";
import { oidcProvider } from ".";

describe("oidc", async () => {
	it("should generate the correct metadata", async () => {
		const auth = betterAuth({
			baseURL: "http://localhost:3000",
			plugins: [oidcProvider()],
		});
		const metadata = await auth.api.getOpenIdConfig();
		expect(metadata).toMatchSnapshot();
	});
});
