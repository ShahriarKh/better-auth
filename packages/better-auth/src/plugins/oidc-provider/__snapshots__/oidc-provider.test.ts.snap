// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`oidc > should generate the correct metadata 1`] = `
{
  "acr_values_supported": [
    "urn:mace:incommon:iap:silver",
    "urn:mace:incommon:iap:bronze",
  ],
  "authorization_endpoint": "http://localhost:3000/api/auth/oauth2/authorize",
  "claims_supported": [
    "sub",
    "iss",
    "aud",
    "exp",
    "nbf",
    "iat",
    "jti",
    "email",
    "email_verified",
    "name",
  ],
  "grant_types_supported": [
    "authorization_code",
  ],
  "id_token_signing_alg_values_supported": [
    "RS256",
    "none",
  ],
  "issuer": "http://localhost:3000",
  "jwks_uri": "http://localhost:3000/api/auth/jwks",
  "registration_endpoint": "http://localhost:3000/api/auth/oauth2/register",
  "response_modes_supported": [
    "query",
  ],
  "response_types_supported": [
    "code",
  ],
  "scopes_supported": [
    "openid",
    "profile",
    "email",
    "offline_access",
  ],
  "subject_types_supported": [
    "public",
  ],
  "token_endpoint": "http://localhost:3000/api/auth/oauth2/token",
  "token_endpoint_auth_methods_supported": [
    "client_secret_basic",
    "client_secret_post",
  ],
  "userInfo_endpoint": "http://localhost:3000/api/auth/oauth2/userinfo",
}
`;
